src/stampit.js

```diff
@@ -21,38 +19,40 @@ import {
 } from 'supermixer';
 
 const create = Object.create;
+function isThenable(value) {
+  return value && isFunction(value.then);
+}
 
 function extractFunctions(...args) {
+  const result = [];
   if (isFunction(args[0])) {
-    return map(args, fn => {
+    forEach(args, fn => { // assuming all the arguments are functions
       if (isFunction(fn)) {
-        return fn;
+        result.push(fn);
       }
     });
   } else if (isObject(args[0])) {
-    const arr = [];
     forEach(args, obj => {
-      forOwn(obj, fn => {
+      forEach(obj, fn => {
         if (isFunction(fn)) {
-          arr.push(fn);
+          result.push(fn);
         }
       });
     });
-    return arr;
   }
-  return [];
+  return result;
 }
 
 function addMethods(fixed, ...methods) {
   return mixinFunctions(fixed.methods, ...methods);
 }
 function addRefs(fixed, ...refs) {
-  fixed.refs = fixed.state = mixin(fixed.refs || fixed.state, ...refs);
+  fixed.refs = fixed.state = mixin(fixed.refs, ...refs);
   return fixed.refs;
 }
 function addInit(fixed, ...inits) {
   const extractedInits = extractFunctions(...inits);
-  fixed.init = fixed.enclose = (fixed.init || fixed.enclose).concat(extractedInits);
+  fixed.init = fixed.enclose = fixed.init.concat(extractedInits);
```

Comment:
Hmm, are you confusing spread and rest? `...` in this case is spread, which is just sugar for concat.

```js
let arr1 = [0, 1, 2];
let arr2 = [3, 4, 5];

let arr3 = [...arr1, ...arr2];
```

is equal to this ES5

```js
var arr1 = [0, 1, 2];
var arr2 = [3, 4, 5];

var arr3 = [].concat(arr1, arr2);
```